addons/mail/static/src/env/env.js

```diff
@@ -0,0 +1,104 @@
+odoo.define('mail/static/src/env/env.js', function (require) {
+'use strict';
+
+const ModelManager = require('mail/static/src/model/model_manager.js');
+
+const env = require('web.commonEnv');
+
+const { Store } = owl;
+const { EventBus } = owl.core;
+
+/**
+ * Messaging store
+ */
+const store = new Store({
+    env,
+    state: {},
+});
+
+/**
+ * Registry of models.
+ */
+env.models = {};
+/**
+ * Environment keys used in messaging.
+ */
+Object.assign(env, {
```

Comment:
The way env is designed in `web` with `web.env` and `web.test_env` dictates how we have to design extensions of the `env` with messaging stuffs. Especially `web.test_env`, which is a complete mess because this is a generator of `env` that is called inside several `test_utils` functions like `addMockEnvironment`.

Because of this, I decided to follow their convention for `web.env` and just put in our `mail/static/src/env/env.js` file code that is in prod (this code is nice as it is and easy to maintain/understand). As for the tests, I've decided to also make them simple extensions of env by means of helper functions like `addMessagingToEnv`, which kind of populates the `env` that is passed to `makeTestEnvironment` env generator in `web.test_env`.

Again, some code duplication, which is a bit annoying, but in the end I think this is the better approach to keep our files maintainable with current messy way of handling `env` in Odoo repository...